cmd/gaia/app/sim_test.go

```diff
@@ -56,6 +59,21 @@ func init() {
 }
 
 func appStateFn(r *rand.Rand, accs []simulation.Account, genesisTimestamp time.Time) json.RawMessage {
+
+	if genesisFile != "" {
+		var genesis tmtypes.GenesisDoc
+		cdc := MakeCodec()
+		bytes, err := ioutil.ReadFile(genesisFile)
+		if err != nil {
+			panic(err)
+		}
+		if err = cdc.UnmarshalJSON(bytes, &genesis); err != nil {
+			panic(err)
+		}
+		fmt.Printf("app state: %s\n", json.RawMessage(genesis.AppState))
+		return json.RawMessage(genesis.AppState)
```

Comment:
I think we ought to break out these two main code paths into separate functions:

```
func appStateFn (...) ...{
    if genesisFile != "" {
        appStateFromGenesis()
    } else {
         appStateCreateGenesis()
    }
}